container/typings/LuigiCompoundContainer.svelte.d.ts

```diff
@@ -11,12 +7,17 @@ export declare interface WebComponentSettings {
  * Luigi Compound Container provides the possibility to insert multiple webcomponent-based microfrontends in one container
  */
 export default class LuigiCompoundContainer extends HTMLElement {
+  /**
+   * The URL used for the renderer
+   */
+  viewurl: string;
+
   /**
    * The configuration for the compound microfrontend
    * Take a look at https://docs.luigi-project.io/docs/navigation-parameters-reference/?section=compound
    * 
    */
-  compoundConfig: any;
+  compoundConfig: Object;
```

Comment:
According to the TS docs, this might not be what we want:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object

`object` seems to be the recommendation, meaning "anything that is not a primitive".

Same could be valid for the other uses of `Object` in this file.

Reply:
Yes I think I know what you refer to, but I believe in this context code is not directly depending on TS typings. Main aim of this file is to help with Documentation Generation. Changing Object -> object only changes the O from Capital to lowercase: 

![Screenshot 2024-01-12 at 11 10 35](https://github.com/SAP/luigi/assets/18335489/94a2d4be-52b1-4e29-af64-b6f4bd1ca7fc)

The context as to what the **object** is we leave to the developer, as we don’t want to enforce such a strict opinionated typescript typings for docu. (for which devs might even have their other own opinions)

Also all of our docu is built upon **Object** notation and that has never caused a misunderstanding. 
(ex.: https://docs.luigi-project.io/docs/luigi-core-api/?section=parameters ). Changing Object-object here would cause a asymmetry with rest of docu. 

The context in which Object->object change would be relevant would be if we need to have super strict opinionated  typings enforced for some functionality reason, for which we would probably also change code or flood the console with warnings (Given object does not meet requirements), which we currently are against doing.

Also, as a historical background, this `.d.ts` file here is primarily just an extra (duplicate) file that’s used for generating documentation and Code Editor suggestions for Typescript users of container package at most. We created and curate these **.d.ts** files manually since generating TS type declarations from our Svelte based container code was too complicated and not possible at the time.

+, it doesn’t affect the functionality whatsoever, as that is left to Svelte code on LuigiContainer.svelte. So Object vs object here would produce no runtime/build errors. In even more depth, the type of these variables boils down to a Javascript Object notation ( check [here](https://github.com/SAP/luigi/blob/main/container/src/LuigiCompoundContainer.svelte#L9)  , so below Typescript territory